stumpy/core.py

```diff
@@ -3852,8 +3860,29 @@ def _compute_P_ABBA(T_A, T_B, m, P_ABBA, mp_func, client=None, device_id=None):
     n_A = T_A.shape[0]
     partial_mp_func = _get_partial_mp_func(mp_func, client=client, device_id=device_id)
 
-    P_ABBA[: n_A - m + 1] = partial_mp_func(T_A, m, T_B, ignore_trivial=False)[:, 0]
-    P_ABBA[n_A - m + 1 :] = partial_mp_func(T_B, m, T_A, ignore_trivial=False)[:, 0]
+    if inspect.signature(partial_mp_func).parameters.get("normalize") is not None:
+        params = partial_mp_func.keywords
+        T_A_subseq_isconstant = params.get("T_A_subseq_isconstant")
+        T_B_subseq_isconstant = params.get("T_B_subseq_isconstant")
+        P_ABBA[: n_A - m + 1] = partial_mp_func(
+            T_A,
+            m,
+            T_B,
+            ignore_trivial=False,
+            T_A_subseq_isconstant=T_A_subseq_isconstant,
+            T_B_subseq_isconstant=T_B_subseq_isconstant,
+        )[:, 0]
+        P_ABBA[n_A - m + 1 :] = partial_mp_func(
+            T_B,
+            m,
+            T_A,
+            ignore_trivial=False,
+            T_A_subseq_isconstant=T_B_subseq_isconstant,
+            T_B_subseq_isconstant=T_A_subseq_isconstant,
+        )[:, 0]
+    else:
+        P_ABBA[: n_A - m + 1] = partial_mp_func(T_A, m, T_B, ignore_trivial=False)[:, 0]
+        P_ABBA[n_A - m + 1 :] = partial_mp_func(T_B, m, T_A, ignore_trivial=False)[:, 0]
 
 
 def _mpdist(
```

Comment:
1. I think the `mp_func` currently for `_mpdist` should be a `functools.partial` because we want to wrap all of the `T_x_subseq_isconstant` inside of that `mp_func`. Thus, I think we should just call the `_mpdist` input parameter as `partial_mp_func` instead of `mp_func`. For example, regardless of whether it is `stump`, `stumped`, or `gpu_stump`, we want all `T_x_subseq_isconstant` to be wrapped inside of it as a `functools.partial` and then used as input for `_mpdist`
2. Inside of `_mpdist`, we'd pass this partial function into `_compute_P_ABBA`, which then further wraps the partial function a second time so that the function adds additional relevant parameters (e.g., `client`, `device_id`). 
3. Inside of `_get_partial_mp_func`, I would check if the `mp_func` is already a partial function. If not, then wrap it. Otherwise, `partial_mp_func = mp_func`:

```
    if client is not None:
        partial_mp_func = functools.partial(mp_func, client)
    elif device_id is not None:
        partial_mp_func = functools.partial(mp_func, device_id=device_id)
    elif type(mp_func) != functools.partial:
        partial_mp_func = functools.partial(mp_func)
    else:
        partial_mp_func = mp_func

    return partial_mp_func
```

Reply:
> 1. I think the `mp_func` currently for `_mpdist` should be a `functools.partial` because we want to wrap all of the `T_x_subseq_isconstant` inside of that `mp_func`. Thus, I think we should just call the `_mpdist` input parameter as `partial_mp_func`

Good point! I will revise the code.